assets/dev/js/editor/elements/views/container.js

```diff
@@ -468,6 +488,73 @@ const ContainerView = BaseElementView.extend( {
 			this.$el.html5Droppable( this.getDroppableOptions() );
 		}
 	},
+
+	handleGridEmptyView() {
+		const currentContainer = this.getCorrectContainerElement();
+
+		this.moveElementToLastChild(
+			currentContainer,
+			currentContainer.find( '> .elementor-empty-view' ),
+		);
+	},
+
+	moveElementToLastChild( parentWrapperElement, childElementToMove ) {
+		let parent = parentWrapperElement.get( 0 ),
+			child = childElementToMove.get( 0 );
+
+		if ( ! parent || ! child ) {
+			return;
+		}
+
+		if ( parent.lastChild === child ) {
+			return;
+		}
+
+		parent.appendChild( child );
+	},
+
+	getCorrectContainerElement() {
+		return this.isBoxedWidth()
+			? this.$el.find( '> .e-con-inner' )
+			: this.$el;
+	},
+
+	shouldIncrementIndex( side ) {
+		if ( this.isFlexContainer() && this.draggingOnBottomOrRightSide( side ) ) {
+			return true;
+		}
+
+		return this.draggingOnBottomOrRightSide( side ) &&
+			! ( this.isGridContainer() && this.emptyViewIsCurrentlyBeingDraggedOver() );
+	},
+
+	draggingOnBottomOrRightSide( side ) {
+		return [ 'bottom', 'right' ].includes( side );
+	},
+
+	isGridContainer() {
+		return 'grid' === this.getContainer().settings.get( 'container_type' );
+	},
+
+	isFlexContainer() {
+		return 'flex' === this.getContainer().settings.get( 'container_type' );
+	},
+
+	isBoxedWidth() {
+		return 'boxed' === this.getContainer().settings.get( 'content_width' );
+	},
+
+	emptyViewIsCurrentlyBeingDraggedOver() {
+		return this.getCorrectContainerElement().find( '> .elementor-empty-view > .elementor-first-add.elementor-html5dnd-current-element' ).length > 0;
```

Comment:
@udidol `elementor.channels.editor.request` seems to run only when the widget is dropped (too late). The best way I could find is by checking if the element has the class `elementor-html5dnd-current-element` because `html5Droppable` gives this class the the current element.

https://github.com/elementor/elementor/blob/211cadf3999d7e28900d51fd23155131af22f2ca/assets/dev/js/editor/elements/views/container.js#L144